meson.build

```diff
@@ -0,0 +1,470 @@
+    config_h.set('HAVE_RENAMEAT2', 1)
+endif
+
+dep_glib = dependency('glib-2.0')
+dep_gio = dependency('gio-2.0')
+dep_gio_unix = dependency('gio-unix-2.0')
+dep_json_glib = dependency('json-glib-1.0')
+dep_fuse3 = dependency('fuse3', version: '>= 3.10.0')
+dep_geoclue = dependency('libgeoclue-2.0',
+                         version: '>= 2.5.2',
+                         required: get_option('geoclue'))
+dep_libportal = dependency('libportal',
+                           required: get_option('libportal'))
+dep_pipewire = dependency('libpipewire-0.3',
+                          version: '>= 0.2.90',
+                          required: get_option('pipewire'))
+
+prog_bwrap = find_program('bwrap')
+config_h.set_quoted('BWRAP', prog_bwrap.full_path())
+
+if dep_glib.version().version_compare('>= 2.60')
+    config_h.set('GLIB_VERSION_MIN_REQUIRED', 'GLIB_VERSION_2_60',
+                 description: 'Ignore massive GTimeVal deprecation warnings in 2.62')
+endif
+
+have_libportal = dep_libportal.found()
+if have_libportal
+    config_h.set('HAVE_LIBPORTAL', 1)
+endif
+
+have_geoclue = dep_geoclue.found()
+if have_geoclue
+    config_h.set('HAVE_GEOCLUE', 1)
+endif
+
+have_pipewire = dep_pipewire.found()
+if have_pipewire
+    config_h.set('HAVE_PIPEWIRE', 1)
+endif
+
+build_docbook = false
+prog_xmlto = find_program('xmlto', required: get_option('docbook-docs'))
+if not get_option('docbook-docs').disabled() and prog_xmlto.found()
+    fs = import ('fs')
+    # we're going to copy this file in to our build tree
+    if fs.is_file(dir_flatpak_intf / 'org.freedesktop.portal.Flatpak.xml')
+        build_docbook = true
+    elif get_option('docbook-docs').enabled()
+        error('Flatpak development files are required to build DocBook docs')
+    endif
+endif
+
+###### systemd units, dbus service files, pkgconfig
+
+base_config = configuration_data()
+base_config.set('prefix', dir_prefix)
+base_config.set('datadir', dir_datadir)
+base_config.set('datarootdir', dir_dataroot)
+base_config.set('libexecdir', dir_libexecdir)
+base_config.set('VERSION', meson.project_version())
+base_config.set('extraargs', '')
+
+systemd_unit_files = files(
+    'src/xdg-desktop-portal.service.in',
+    'document-portal/xdg-permission-store.service.in',
+    'document-portal/xdg-document-portal.service.in',
+
+)
+foreach unit: systemd_unit_files
+    configure_file(input: unit,
+        output: '@BASENAME@',
+        configuration: base_config,
+        install: true,
+        install_dir: dir_systemd_userunit,
+        )
+endforeach
+
+doc_portal_service_file = files('document-portal/org.freedesktop.portal.Documents.service.in')
+permission_portal_service_file = files('document-portal/org.freedesktop.impl.portal.PermissionStore.service.in')
+desktop_portal_service_file = files('src/org.freedesktop.portal.Desktop.service.in')
+
+dbus_service_files = [
+    doc_portal_service_file,
+    permission_portal_service_file,
+    desktop_portal_service_file,
+]
```

Comment:
Why do these each have their own variable which is never used other than to treat as opaque members of an array?

Also note that you can pass strings directly to configure_file, the point of a files() object is to remember which directory it can be found in *iff* you are going to use it from a meson.build in another directory.

Reply:
the first two are used in `tests/services/meson.build`, hence the files object so I don't have to mangle the directories. The third one is indeed unused but added for consistency.

This has partially changed now since the two `document-portal` files are now in the `document-portal/meson.build` file, the `Desktop.service` file is now just a string since there's no group for consistency anymore.